services/thangs_sync_service.py

```diff
@@ -234,11 +236,11 @@ def get_optional_value(key: str):
         except requests.HTTPError as e:
             print(str(e))
             if e.response.status_code == 401:
-                self.__login_service.login_user()
+                self.__login_service.login_user(get_threading_service().wrap_up_threads)
                 self.__sync_current_blender_file()
             elif e.response.status_code == 403:
                 self.remove_sync_info_text_block()
-                self.__sync_current_blender_file()
+                self.__clear_ui_status_message()
```

Comment:
The `Deny` button on the access grant is giving us a 403. When a user hit `Deny` it would reopen the access grant page over and over which due to the `self.__sync_current_blender_file()` line. We may need to decipher if the 403 is from access grant or another source. If it's another source retry.

Reply:
Maybe if we get a 403 we throw some kind of exception we can catch and abort whatever we were doing?